plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergNodeLocalDynamicSplitPruning.java

```diff
@@ -0,0 +1,247 @@
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.connector.DynamicFilter;
+import io.trino.spi.predicate.Domain;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.type.Type;
+import io.trino.testing.TestingConnectorSession;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SchemaParser;
+import org.apache.iceberg.types.Types;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import static io.trino.orc.metadata.CompressionKind.NONE;
+import static io.trino.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.trino.plugin.hive.HiveType.HIVE_INT;
+import static io.trino.plugin.hive.HiveType.HIVE_STRING;
+import static io.trino.plugin.iceberg.IcebergFileFormat.ORC;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static io.trino.type.InternalTypeManager.TESTING_TYPE_MANAGER;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+
+public class TestIcebergNodeLocalDynamicSplitPruning
+{
+    private static final String ICEBERG_CATALOG_NAME = "iceberg";
+    private static final String SCHEMA_NAME = "test";
+    private static final String TABLE_NAME = "test";
+    private static final Column PARTITION_COLUMN = new Column("l_partkey", HIVE_INT, Optional.empty());
+    private static final ColumnIdentity PARTITION_COLUMN_IDENTITY = new ColumnIdentity(1, PARTITION_COLUMN.getName(), ColumnIdentity.TypeCategory.PRIMITIVE, ImmutableList.of());
+    private static final IcebergColumnHandle PARTITION_ICEBERG_COLUMN_HANDLE = new IcebergColumnHandle(PARTITION_COLUMN_IDENTITY, INTEGER, ImmutableList.of(), INTEGER, Optional.empty());
+    private static final int PARTITION_COLUMN_VALUE = 42;
+    private static final Column DATA_COLUMN = new Column("col1", HIVE_STRING, Optional.empty());
+    private static final ColumnIdentity DATA_COLUMN_IDENTITY = new ColumnIdentity(2, DATA_COLUMN.getName(), ColumnIdentity.TypeCategory.PRIMITIVE, ImmutableList.of());
+    private static final IcebergColumnHandle DATA_ICEBERG_COLUMN_HANDLE = new IcebergColumnHandle(DATA_COLUMN_IDENTITY, VARCHAR, ImmutableList.of(), VARCHAR, Optional.empty());
+    private static final String DATA_COLUMN_VALUE = "hello world";
+    private static final Schema TABLE_SCHEMA = new Schema(
+            optional(PARTITION_COLUMN_IDENTITY.getId(), PARTITION_COLUMN.getName(), Types.IntegerType.get()),
```

Comment:
There's no need for this to be a partitioned column, we will be able to prune splits even with DF on a data column

Reply:
removed all references to "partition" which may be indeed misleading for the functionality tested.